marlowe-playground-client/src/Marlowe/Linter.purs

```diff
@@ -452,7 +452,12 @@ lintValue env t@(Term (Scale h@(Hole _ _ _) c) pos) = do
 
 lintValue env t@(Term (ChoiceValue choiceId a) pos) = do
   modifying _holes (getHoles choiceId)
-  pure (ValueSimp pos false t)
+  sa <- lintValue env a
+  case sa of
+    ConstantSimp _ _ c1 -> pure (ConstantSimp pos true (-c1))
+    _ -> do
+      markSimplification constToVal SimplifiableValue a sa
+      pure (ValueSimp pos false t)
```

Comment:
This is not right, it only works for `NegValue`. But it wasn't right before either, good point. It should always be:
```
sa <- lintValue env a
markSimplification constToVal SimplifiableValue a sa
pure (ValueSimp pos false t)
```
I think. No need to pattern match in `sa`. Maybe I should document this

Reply:
Ahhh, sorry I misunderstood what was going on here.